operator/standalone/controller.go

```diff
@@ -89,14 +89,20 @@ func (r *PostgresStandaloneReconciler) CreateDeployment(ctx context.Context, ins
 	return reconcile.Result{RequeueAfter: 10 * time.Second}, err
 }
 
-// Delete prepares the given instance for deletion.
-func (r *PostgresStandaloneReconciler) Delete(ctx context.Context, instance *v1alpha1.PostgresqlStandalone) (reconcile.Result, error) {
+// DeleteDeployment prepares the given instance for deletion.
+func (r *PostgresStandaloneReconciler) DeleteDeployment(ctx context.Context, instance *v1alpha1.PostgresqlStandalone) (reconcile.Result, error) {
 	// we don't need to delete it by ourselves, since the deletion timestamp is already set.
 	// Just remove all finalizers
 	log := ctrl.LoggerFrom(ctx)
+	d := NewDeleteStandalonePipeline(r.client, instance)
+	log.Info("Deleting instance")
+	err := d.RunPipeline(ctx)
+	if err != nil {
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, err
+	}
 	controllerutil.RemoveFinalizer(instance, finalizer)
 	log.Info("Deleting")
-	err := r.client.Update(ctx, instance)
+	err = r.client.Update(ctx, instance)
 	return reconcile.Result{}, err
```

Comment:
I propose to move the finalizer and updating the recource as part of the pipeline

Reply:
That's ok, we can move it into the forth step.